### DISCUSSION TITLE
Setting device_class to window for Aqara MCCGQ11LM removes sensors in HA

### DISCUSSION THREAD

**Discussion by mcarson75** [Koenkk/zigbee2mqtt#20723] [Devices]
I have several Aqara MCCGQ11LM contact sensors on windows around my house, and I noticed recently that the battery is shown as "Unavailable" in all of them in HA, despite the necessary info being in MQTT and Z2M.  In trying to fix this, I found that the only way to get these sensors back was to delete/re-pair the device in Z2M.

Upon doing this, I saw that this was configured as a door instead of a window in HA.  I remembered then that I had set `device_class: window` for the sensors in configuration.yaml.  When I added this to the configuration, I didn't see a change in HA, even after a restart of Z2M.  I renamed the device in Z2M and selected "Update Home Assistant entity id."  On doing this, it removed every sensor except for the contact sensor from HA.  After this I tried removing the `device_class` from the yaml and changing the name again, but the sensors did not reappear.  I see many messages like this in the HA logs:

```
Error 'expected SensorDeviceClass or one of 'date', 'enum', 'timestamp', 'apparent_power', 'aqi', 'atmospheric_pressure', 'battery', 'carbon_monoxide', 'carbon_dioxide', 'current', 'data_rate', 'data_size', 'distance', 'duration', 'energy', 'energy_storage', 'frequency', 'gas', 'humidity', 'illuminance', 'irradiance', 'moisture', 'monetary', 'nitrogen_dioxide', 'nitrogen_monoxide', 'nitrous_oxide', 'ozone', 'ph', 'pm1', 'pm10', 'pm25', 'power_factor', 'power', 'precipitation', 'precipitation_intensity', 'pressure', 'reactive_power', 'signal_strength', 'sound_pressure', 'speed', 'sulphur_dioxide', 'temperature', 'volatile_organic_compounds', 'volatile_organic_compounds_parts', 'voltage', 'volume', 'volume_storage', 'water', 'weight', 'wind_speed' for dictionary value @ data['device_class']' when processing MQTT discovery message topic: 'homeassistant/sensor/0x00158d000911d70a/battery/config', message: '{'availability': [{'topic': 'zigbee2mqtt/bridge/state', 'value_template': '{{ value_json.state }}'}], 'device': {'identifiers': ['zigbee2mqtt_0x00158d000911d70a'], 'manufacturer': 'Xiaomi', 'model': 'Aqara door & window contact sensor (MCCGQ11LM)', 'name': 'Garage Window (Left)', 'sw_version': '3000-0001', 'via_device': 'zigbee2mqtt_bridge_0x00124b00257b161d'}, 'device_class': 'window', 'enabled_by_default': True, 'entity_category': 'diagnostic', 'object_id': 'garage_window_(left)_battery', 'origin': {'name': 'Zigbee2MQTT', 'sw_version': '1.35.1', 'support_url': 'https://www.zigbee2mqtt.io'}, 'state_class': 'measurement', 'state_topic': 'zigbee2mqtt/Garage Window (Left)', 'unique_id': '0x00158d000911d70a_battery_zigbee2mqtt', 'unit_of_measurement': '%', 'value_template': '{{ value_json.battery }}', 'platform': 'mqtt'}'
```

How can I configure this so that it is defined as a window and also still get all of the additional sensors as well?

**Comment by tteck** [comment]
Use the "show as" feature in Home Assistant.

![Screenshot 2024-01-10 12 20 45 PM](https://github.com/Koenkk/zigbee2mqtt/assets/21342670/e6fc32d2-23b6-4b6d-8903-5d3d4bb91482)

**Reply by mcarson75** [reply]
I've tried that but it doesn't work as expected.  It updates the icon but not the name:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/c3a2b7db-15e3-4c38-b973-88d81e0fb535)

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/2c840bef-09c9-4ca3-92b9-e14bfbf89e68)

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/a9e63f6d-cc2a-465b-ac19-592a28d3d734)

**Reply by tteck** [reply]
Of course you have to change the name. Currently you have given no name, so it uses the sensor name.

**Reply by mcarson75** [reply]
Fair point, that I can change the name, but if I change `device_class` in Z2M it names it correctly.  I would be nice to be able to configure the device class in Z2M so I didn't have to modify every one in HA - thus my question.

**Reply by tteck** [reply]
![Screenshot 2024-01-10 1 06 02 PM](https://github.com/Koenkk/zigbee2mqtt/assets/21342670/292d5f98-87fa-45ad-a830-5ec4f6c4ead7)

**Reply by mcarson75** [reply]
I'm not sure what this is supposed to reference?  The name there does not have anything to do with whether the contact label is Door or Window.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/df87a284-e2be-48cc-886b-8a532feb46fc)

**Reply by tteck** [reply]
Naming it correctly will.

**Reply by tteck** [reply]
Just name it "Right Office Window".

**Reply by mcarson75** [reply]
How is this set incorrectly?  But to test it out I set the name in HA instead of letting Z2M set it.  No change to the contact label.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/75d115fb-caa1-40d0-8684-7bab819b2ae3)

**Reply by tteck** [reply]
hit update

**Reply by mcarson75** [reply]
Um yes, I hit update and then went back in to the screen for the screenshot.  There is no change.

Just to prove it, I renamed it differently, HIT UPDATE (see the name in the upper left), told it to update all of the entity names, and THEN went back in and took this screenshot.  It's still a door.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/ad8f739f-59e2-49bf-b919-97622e1ed868)

**Reply by tteck** [reply]
Then rename "door"

**Reply by tteck** [reply]
You should have given it a "name" when changing the "show as"

**Reply by mcarson75** [reply]
As mentioned above in reference to changing the entity name: "Fair point, that I can change the name, but if I change `device_class` in Z2M it names it correctly. I would be nice to be able to configure the device class in Z2M so I didn't have to modify every one in HA - thus my question."

To elaborate, if I set the device_class in Z2M it changes the contact name to simply "Window" but then the rest of the sensors (battery, voltage, temperature) are deleted from HA.

I would rather not have to configure this in two different places (HA and Z2M) - thus my initial question.

In addition, if I leave the name as default then for the contact name it just shows "door" but when I go into "More Info" it shows which one.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/48053c2e-a293-403d-a3d7-5dd87a319ab4)
![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/a7970dd2-16dd-405c-a0c2-266462d277b6)

If I rename the entity, then it only shows the new entity name as both the contact name and in More Info, so I have to choose whether to have it just "Window" (which I prefer) or to name it the full name (which then shows the full name just for the contact).

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/8dbd849a-fab1-4092-aa61-397347b0602a)
![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/acc71b8b-cc12-4398-a8d4-d91211f4975b)

**Reply by tteck** [reply]
I believe _most_ people prefer their entity name to accurately represent the specific entity (e.g., 'Bathroom Window') rather than a generic term ('Window'). It's more convenient for users not to have to click to discern which window is being referred to.

**Reply by mcarson75** [reply]
It's a fair point, I know it's personal preference.  I guess to clarify - on my other sensors (which currently have `device_class` set to `window` in Z2M) the entity "Name" is the full name (`Living Room Window (Left)`) but when I go into the Device itself then the binary sensor is just `Window`.  This seems clean and logical to me.  However, if I manually modify the entity name then on the device page it will have the full name instead of the short name.  Not sure how to configure it to show the short name only in the Device?

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/c3495bbe-e0b6-4972-98c9-7f2b523a5975)
![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/1af105de-7e6e-4606-8b0f-0713b7fbad1c)

**Comment by mcarson75** [comment]
Whoops, I put a screenshot of the wrong entity.  This is the right one.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/54790846/a91c68a9-2884-4e00-af0a-75f28a42eb1b)
